extensions/performance/src/main/java/io/deephaven/engine/table/impl/util/PerformanceQueriesGeneral.java

```diff
@@ -1,53 +1,37 @@
 /**
- * Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+ * Copyright (c) 2016-2023 Deephaven Data Labs and Patent Pending
  */
 package io.deephaven.engine.table.impl.util;
 
 import io.deephaven.engine.table.Table;
-
+import io.deephaven.plot.Figure;
+import io.deephaven.plot.PlottingConvenience;
 import io.deephaven.util.QueryConstants;
-import io.deephaven.util.annotations.ScriptApi;
 
 import java.util.Arrays;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.OptionalLong;
 
-import static io.deephaven.api.agg.Aggregation.*;
-
-public class PerformanceQueries {
-    private static final boolean formatPctColumns = true;
-
-    /**
-     * <p>
-     * Takes in a query evaluation number and returns a view for that query's performance data.
-     * </p>
-     *
-     * <p>
-     * You can obtain query evaluation numbers, which uniquely identify a query and its subqueries, via the
-     * {@code QueryPerformance} and {@code QueryOperationPerformance} tables, calling
-     * {@code TableLoggers.queryPerformanceLog()} or {@code TableLoggers.queryOperationPerformanceLog()}.
-     * </p>
-     *
-     * <p>
-     * The query performance log contains data on how long each query takes to run. Examples of what constitutes one
-     * individual query, for performance logging purposes, include:
-     * <ul>
-     * <li>A new command in the console (i.e. type something, then press the return key)</li>
-     * <li>A sort, filter, or custom column generated by a UI</li>
-     * <li>A call from a client API external application</li>
-     * </ul>
-     * </p>
-     *
-     * @param evaluationNumber evaluation number
-     * @return query performance table.
-     */
-    @ScriptApi
-    public static Table queryPerformance(final long evaluationNumber) {
+import static io.deephaven.api.agg.Aggregation.AggFirst;
+import static io.deephaven.api.agg.Aggregation.AggMax;
+import static io.deephaven.api.agg.Aggregation.AggPct;
+import static io.deephaven.api.agg.Aggregation.AggSum;
+
+/**
+ * Generalizes {@link PerformanceQueries} to accept table parameters and make evaluation number parameter optional.
+ */
+public class PerformanceQueriesGeneral {
+    private static boolean formatPctColumns = true;
```

Comment:
Is your intention to set this to `false` in DHE, or in specific cases? It's a bit fragile. Have you considered adding a boolean parameter to all of the methods?

It looks like a few of the methods have formatPctColumnsLocal; what is the global static for?

Reply:
I decided not to set `formatPctColumns` from true to false to true as needed. Fwiw, this static field already existed despite never being set to anything other than initial value.

Instead, I added `formatPctColumnsLocal` where it was _needed_ (meaning to avoid the bug referenced in N2 via good practices reg order of operations). In this case, both the local and static value need to be true to allow formatting. I don't know the purpose of static `formatPctColumns`, but did not want to remove it w/o a DHC dev's say-so.